### PR TITLE
Common (CPU, GPU) changes to directory structure for resampling.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

### CODE REVIEW

**Comment by mzient** [comment]
Builds: #648780

File: dali/kernels/context.h [hunk 1]
```diff
@@ -43,7 +43,7 @@ class Scratchpad {
   ///        in the memory of type `alloc_type`.
   template <AllocType alloc_type, typename T, size_t dim>
   TensorView<AllocBackend<alloc_type>, T, dim> AllocTensor(TensorShape<dim> shape) {
-    return { Allocate<T>(alloc_type, Volume(shape)), std::move(shape) };
+    return { Allocate<T>(alloc_type, volume(shape)), std::move(shape) };
```
Comment (jantonguirao) [inline dali/kernels/context.h]: Unify Volume/volume into one volume function

File: dali/kernels/tensor_shape.h [hunk 2]
```diff
@@ -316,6 +316,30 @@ TensorShape<shape_cat_ndim(left_ndim, right_ndim)> shape_cat(const TensorShape<l
   return result;
 }
 
+/// @brief Appends a scalar to a shape
+template <int ndim, int out_dim = ndim == DynamicDimensions ? ndim : ndim + 1>
+TensorShape<out_dim> shape_cat(const TensorShape<ndim> &left, int64_t right) {
+  TensorShape<out_dim> result;
+  result.resize(left.size() + 1);
+  for (int i = 0; i < left.size(); i++) {
+    result[i] = left[i];
```
Comment (jantonguirao) [inline dali/kernels/tensor_shape.h]: no copy constructor in TensorShape?

File: dali/kernels/context.h (hunk 1 above)
Comment (mzient) [reply dali/kernels/context.h]: Filed a task to track the issue separately.

File: dali/kernels/tensor_shape.h (hunk 2 above)
Comment (mzient) [reply dali/kernels/tensor_shape.h]: Size differs.

File: dali/kernels/context.h (hunk 1 above)
Comment (mzient) [reply dali/kernels/context.h]: https://github.com/NVIDIA/DALI/pull/559

**Pull request merged**